### ISSUE TITLE
База знаний

### ISSUE THREAD

**Issue by BaldiSlayer** [BaldiSlayer/rofl-lab1#7]
**TLDR**
В комментах под этим `issue` напиши по какому разделу ТФЯ ты будешь готовить теорию

**FULL**

Внимание всем тем, кто выбрал заполнять базу знаний. Предлагаю в этом `issue` разобрать темы. Напоминаю, что каждый должен дать по 12 элементов базы (базу заполняем **теорией**, не задачами). Как понимаю это я: элемент базы знаний это пара ключ значение. Что-то по типу:
```json
{
  "question": "Что такое ТФЯ?",
  "answer": "теория формальных языков",
}
```

**Предполагаемое решение**
Мы создадим в репозитории `json/yaml` файл и в нем будем хранить эти пары ключ значение. 

В итоге у нас получится файл вида:
```json
[
  {
    "question": "Что такое ТФЯ?",
    "answer": "теория формальных языков",
  },
  {
     "question": "Что такое ДКА?",
     "answer": "детерминированный конечный автомат",
  },
]
```

Дальше, при старте бэкенда будем векторизовывать и выгружать векторизованные данные бд, а потом будем искать ближайшие (хотя в принципе в таком тз можно и прям на бэке в "мапе" это хранить, крч хз, обсуждаемо, я больше за БД).

**Comment by BaldiSlayer** [comment]
я с @molivka берем теорию по автоматам

**Comment by Egorrrad** [comment]
Беру основные определения: алфавит, слово, язык, конкатенация, свободный моноид слов и операции над языками

**Comment by NailorSH** [comment]
Беру на себя Операции над множествами

**Comment by nybc4ik** [comment]
Беру леммы о накачке

**Comment by ark2016** [comment]
беру LL-языки, в частности лекции [Охотина ](https://users.math-cs.spbu.ru/~okhotin/teaching/fg_2019/okhotin_fg_2019_l12.pdf)

**Comment by TonitaN** [comment]
Так, прошу учесть, что база знаний должна содержать данные, по которым ваша модель по умолчанию галлюцинирует. Т.е. вы должны сначала проверить по вашему запросу: если LLM и так пишет адекватный ответ ("что такое конечный автомат?" и подобные вопросы - тем более, на англ яз, будет много адекватных ответов), то ищем другой запрос. Галлюцинаций хватит на всех, но поискать их надо.

**Comment by AlisaEee** [comment]
Я беру тему: "автоматы"

**Comment by TonitaN** [comment]
Те, кто делают чат: рекомендую ещё подумать в следующем направлении. Нередко LLM куда более прокаченные на английском, а запросы, скорее всего, всем хочется писать на русском. Нужно посмотреть, что выгоднее: написать собственный запрос на перевод явно, или уповать на то, что внутренние алгоритмы модели сделают это сами, если русского контекста не хватит.

**Comment by seregazakharin** [comment]
Я беру на себя: "Регулярные выражения и языки"

**Comment by HermanDp45** [comment]
> Я беру на себя: "Регулярные выражения и языки"

я вместе с Сергеем эту тему также беру

**Comment by Hexxet** [comment]
Я и Маша Пишикина берём тему - Введение в теорию машин Тьюринга

**Comment by TonitaN** [comment]
Ещё есть прекрасные темы, которые никто не взял:
- системы переписывания термов
- фундированные множества и фундированная индукция (можно - все виды индукции)
- разрешимость и перечислимость (заодно разберётесь - на экзамене этого не будет, но в теоретический минимум в принципе это должно входить, правда, обычно это идёт в логике или в дискре, так что на ТФЯ попадает только самый край)

**Comment by alexgroga** [comment]
Беру контекстно-свободные грамматики и свойства контекстно-свободных языков

**Comment by advanced12iq** [comment]
Беру разрешимость и перечислимость

**Comment by m-tisssh** [comment]
Я и Дима берем неразрешимость ;)

**Comment by TonitaN** [comment]
Группа 52 спрашивала, можно ли заспамить БД примерами (например, ответами на вопрос, эквивалентны ли два регулярных выражения). Ответ: если это вопрос очень базовый, с совсем типичными регулярками, на который модель галлюцинирует, то один-два бывает полезны. Но многочисленные однотипные элементы БД такого вида будут бесполезны уже всем: и вам (не помогут отвечать на вопросы), и мне (не дадут никакого представления, на пояснения к чему стоит обратить больше внимания).

**Comment by pelmenkus** [comment]
Беру средства regex-машин в современных ЯП

**Comment by GekK9** [comment]
Беру системы переписывания термов

**Comment by GoArty** [comment]
Беру средства regex-машин в современных ЯП

**Comment by DSIsakov** [comment]
Беру средства regex-машин в современных ЯП

**Comment by papashik** [comment]
мы с Соней берем контекстно-свободные грамматики и свойства контекстно-свободных языков

**Comment by massssha** [comment]
беру фундированные множества и индукцию

**Comment by ssairu** [comment]
беру алгоритмически разрешимые и неразрешимые проблемы

**Comment by ktveritnev23** [comment]
Беру средства regex-машин в современных ЯП

**Comment by Fitymi54** [comment]
беру контекстно-свободные грамматики и языки
